pints/_mcmc/_slice_stepout.py

```diff
@@ -0,0 +1,684 @@
+            if self._init_bisection:
+                self._l_hat = self._l_bar
+                self._r_hat = self._r_bar
+                self._init_bisection = False
+
+            # Apply bisection to endpoint edges
+            if self._a_bar > 0:
+
+                # Prepare bisection
+                if self._bisection:
+                    self._w_bar = self._w_bar / 2
+                    self._temp_l_bisection = np.array(self._proposed,
+                                                      copy=True)
+                    self._temp_r_bisection = np.array(self._proposed,
+                                                      copy=True)
+                    self._set_l_bisection = True
+                    self._set_r_bisection = True
+                    self._bisection = False
+
+                # Apply bisection to left edge
+                if self._set_l_bisection:
+
+                    self._l_bisection = (self._l_hat + self._w_bar)
+                    self._temp_l_bisection[self._active_param_index] = (
+                        self._l_bisection)
+                    self._ready_for_tell = True
+                    return np.array(self._temp_l_bisection, copy=True)
+
+                # Apply bisection to right edge
+                if self._set_r_bisection:
+                    self._r_bisection = (self._r_hat - self._w_bar)
+                    self._temp_r_bisection[self._active_param_index] = (
+                        self._r_bisection)
+                    self._ready_for_tell = True
+                    return np.array(self._temp_r_bisection, copy=True)
+
+            # Find candidate point by flipping from the current point to
+            # the opposide side
+            self._proposed[self._active_param_index] = (self._l_hat +
+                                                        self._r_hat -
+                                                        self._current
+                                                        [self.
+                                                         _active_param_index])
+            self._ready_for_tell = True
+            return np.array(self._proposed, copy=True)
+
+        else:
+            # Sample new trial point by sampling uniformly from the
+            # interval ``I=(l,r)``
+            u = np.random.uniform()
+            self._proposed[self._active_param_index] = self._l + u * (self._r -
+                                                                      self._l)
+
+            # Send trial point for checks
+            self._ready_for_tell = True
+            return np.array(self._proposed, copy=True)
+
+    def bisection_steps(self):
+        """
+        Returns integer limit overrelaxation endpoint accuracy to
+        ``2^(-bisection steps) * width``.
+        """
+        return self._a
+
+    def current_log_pdf(self):
+        """ See :meth:`SingleChainMCMC.current_log_pdf()`. """
+        return self._current_log_pdf
```

Comment:
Hey @lorcandelaney 

There's something worth thinking about whenever you write these methods, and that is: Will this make it easier for users to (accidentally) break the integrity of my class? I.E. by changing something, and accidentally creating some invalid state.

In this case, because you're returning an array, I can do this:
```
x = sampler.current_log_pdf()
x[2] = 12
```
which updates your internal state.
So its usually better to do ```return np.copy(self._current_log_pdf)```, or to make the object read-only (if you don't need to modify it yourself). That doesn't _prevent_ the user from unsetting the read only flag and modifying it, but at least they'll be doing it on purpose.
Not that for e.g. ```x = 5; return x``` this isn't a problem, because you can't modify an integer. (You could write ```x = method.get_x(); x = 12```, but that would change your local variable to point at a new integer 12, without affecting the x in the class). (Strings are similar, as they're defined as _immutable_: you can't change a string, doing e.g. ```x = '123'; x += '4'``` creates a new string '1234' and points x at it)